lib/backend/k8s/resources/watcher.go

```diff
@@ -0,0 +1,155 @@
+// Copyright (c) 2016-2017 Tigera, Inc. All rights reserved.
+
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package resources
+
+import (
+	"context"
+	"fmt"
+	"sync/atomic"
+
+	log "github.com/sirupsen/logrus"
+	kwatch "k8s.io/apimachinery/pkg/watch"
+
+	"github.com/projectcalico/libcalico-go/lib/backend/api"
+	"github.com/projectcalico/libcalico-go/lib/backend/model"
+	cerrors "github.com/projectcalico/libcalico-go/lib/errors"
+)
+
+const (
+	resultsBufSize = 100
+)
+
+func newK8sWatcherConverter(ctx context.Context, converter ConvertK8sResourceToKVPair, k8sWatch kwatch.Interface) api.WatchInterface {
+	ctx, cancel := context.WithCancel(ctx)
+	return &k8sWatcherConverter{
+		converter:  converter,
```

Comment:
nitty..so feel free to ignore.. but the order of these doesn't match how they're defined in the struct which makes it a bit harder to follow.